### PR TITLE
feat: add requestHandler ctor pass through type

### PR DESCRIPTION
**Pull request by kuhe**
Currently, to override any options in the default request handler, the requestHandler class must be instantiated.

This adds a type representing default request handler ctor params, 
intended to allow the following syntactic sugar:
```js
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { NodeHttpHandler } from "@smithy/node-http-handler";
import https from "https";    

var agent = new https.Agent({
  maxSockets: 25
});

var dynamodbClient = new DynamoDBClient({
  requestHandler: new NodeHttpHandler({
    httpsAgent: agent
 })
});
```

the PR's ultimate goal is to allow the following:

```js
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import https from "https";    

var agent = new https.Agent({
  maxSockets: 25
});

var dynamodbClient = new DynamoDBClient({
  requestHandler: {
    httpsAgent: agent
  }
});
```

This may appear to be a very mild simplification, but it avoids user errors like accidentally installing a locked version of `@smithy/node-http-handler` or needing to differentiate between the previous `@aws-sdk/node-http-handler` and its migrated name.

### CODE REVIEW

File: packages/protocol-http/package.json [hunk 1]
```diff
@@ -16,6 +16,13 @@
   "main": "./dist-cjs/index.js",
   "module": "./dist-es/index.js",
   "types": "./dist-types/index.d.ts",
+  "browser": {
+    "./dist-es/httpHandlerInitialization": "./dist-es/httpHandlerInitialization.browser"
+  },
```
Comment (syall) [inline packages/protocol-http/package.json:21]: does this need a `dist-cjs` entry point?

File: packages/protocol-http/src/httpHandlerInitialization.browser.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Alternate compilation for browser excluding http Agent types.
+ *
+ * @internal
+ */
+export type DefaultRequestHandlerInitializationPassThroughParameters = FetchHttpHandlerOptions;
```
Comment (syall) [inline packages/protocol-http/src/httpHandlerInitialization.browser.ts:6]: should these types and interfaces live in `@smithy/types`, and the `@smithy/*-http-handler` packages import them so there is no duplication?

File: packages/protocol-http/src/httpHandlerInitialization.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * This type represents an alternate client constructor option for the entry
+ * "requestHandler". Instead of providing an instance of a requestHandler, the user
+ * may provide the requestHandler's constructor options for either the
+ * NodeHttpHandler or FetchHttpHandler.
+ *
+ * For other RequestHandlers, constructor parameter passthrough is not available.
+ *
+ * @public
+ */
+export type RequestHandlerParams = NodeHttpHandlerOptions | FetchHttpHandlerOptions;
```
Comment (syall) [inline packages/protocol-http/src/httpHandlerInitialization.ts:15]: should these types and interfaces live in `@smithy/types`, and the `@smithy/*-http-handler` packages import them so there is no duplication?

File: packages/protocol-http/src/httpHandlerInitialization.browser.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Alternate compilation for browser excluding http Agent types.
+ *
+ * @internal
```
Comment (syall) [inline packages/protocol-http/src/httpHandlerInitialization.browser.ts:4]: the browser one is marked as `@internal`, but the Node.js export is marked as `@public`

**Review by syall — CHANGES_REQUESTED** [review]

File: packages/protocol-http/package.json (hunk 1 above)
Comment (kuhe) [reply packages/protocol-http/package.json]: no, I'm going to remove this

File: packages/protocol-http/src/httpHandlerInitialization.ts (hunk 3 above)
Comment (kuhe) [reply packages/protocol-http/src/httpHandlerInitialization.ts]: yes, let's do that

File: packages/protocol-http/src/httpHandlerInitialization.browser.ts (hunk 4 above)
Comment (kuhe) [reply packages/protocol-http/src/httpHandlerInitialization.browser.ts]: removed the extra browser file after the move to `@smithy/types`

**Review by syall — APPROVED** [review]

**Pull request merged**